bundles/paikkatietoikkuna/coordinatetransformation/components/CoordinateSystemSelection.js

```diff
@@ -0,0 +1,232 @@
+        },
+        /**
+         * @method populateSelect
+         * @desc creates an instance of the { Oskari.coordinatetransformation.component.select },
+         * and fills it with data
+         */
+        populateSelect: function () {
+            var me = this;
+            var wrapper = this.getElement();
+            this.select.create();
+            this.selectInstance = this.select.getSelectInstances();
+            this.dropdowns = this.select.getDropdowns();
+            Object.keys( this.dropdowns ).forEach( function( key ) {
+                jQuery( wrapper.find( '.transformation-system' ).find( me.makeClassSelector(key) ).append( me.dropdowns[key] ));
+            });
+            
+            me.getElement().find('.system').each(function (index) {
+                jQuery( this ).on('change', function ( e ) {
+                    me.handleSystemChange(e);
+                });
+            })
+        },
+        handleInfoLink: function () {
+            var me = this;
+            this.getElement().find('.infolink').on('click', function ( event ) {
+                event.stopPropagation();
+                var key = this.parentElement.parentElement.dataset.system;
+                me.systemInfo.show( jQuery( this ), key );
+            });
+        },
+        handleSystemChange: function (e) {
+            var current = this.selectInstance[e.currentTarget.dataset.system];
+            var currentValue = current.getValue();
+            this.handleSelectValueChange(currentValue);
+        },
+        /**
+         * @method updateDropdownOptions
+         * @param {string} valueClass - class selector to show options for 
+         * @param {string} keyToEmpty - optional param to empty only one specific key in the dropdown 
+         */
+        updateDropdownOptions: function (dropdownId, valueId) {
+            if ( typeof valueId === 'string' ) {
+                this.dropdowns[dropdownId].find( 'option' ).hide();
+                this.dropdowns[dropdownId].find( valueId ).show();
+            } else {
+                this.dropdowns[dropdownId].find( 'option' ).show();
+            }
+        },
+        makeClassSelector: function (variable) {
+            return "." + variable;
+        },
+        /**
+         * @method handleSelectValueChange
+         * @param {string} currentValue - value of the dropdown we changed
+         * @desc handle hiding and showing dropdown options based on user selection 
+         */
+        handleSelectValueChange: function (currentValue) {
+            var me = this;
+            var dropdowns = this.dropdowns;
+            var instances = this.selectInstance;
+            var datum = instances.datum.getValue();
+            var coordinate = instances.coordinate.getValue();
+            var projection = instances.projection.getValue();
+            var clsSelector = this.makeClassSelector;
+
+            var table;
+            var system = this.element.attr('data-system');
```

Comment:
is this.element the wrapper for the whole ui or something else here? Does it gain the data-system attribute somewhere?

Reply:
this.element is the wrapper for the coordinate system selection dropdowns, it gains the data-system attribute on creation (https://github.com/nxholm/oskari-frontend/blob/coordtrans/functionality-fixes/bundles/paikkatietoikkuna/coordinatetransformation/view/transformation.js#L73)